src/main/java/core/characters/CharacterManager.java

```diff
@@ -1,80 +1,137 @@
+    public CharacterManager() {
+        this.characterEntities = new HashMap<Integer, GameCharacter>();
+        this.keylist = new ArrayList<Integer>();
     }
 
-    public void moveCharacter() {
-        // Updates the characters position
+    public void addCharacter(GameCharacter character) {
+        /*
+        * Player will always be hashed as 0
+        * Defenders will be hashed as even ints
+        * Enemies will be hashed as odd ints
+        * TODO: Refactor
+        * */
+        int maxElement = this.keylist.get(this.keylist.size() - 1);
+
+        if (Objects.equals(character.team, "player")) {
+            this.characterEntities.put(0, character);
+            this.keylist.add(0);
+        } else if (Objects.equals(character.team, "defender")) {
+            if (maxElement % 2 == 0) {
+                this.characterEntities.put(maxElement + 2, character);
+                this.keylist.add(maxElement + 2);
+            } else {
+                this.characterEntities.put(maxElement + 1, character);
+                this.keylist.add(maxElement + 1);
+            }
+        } else {
+            if (maxElement % 2 == 0) {
+                this.characterEntities.put(maxElement + 1, character);
+                this.keylist.add(maxElement + 1);
+            } else {
+                this.characterEntities.put(maxElement + 2, character);
+                this.keylist.add(maxElement + 2);
+            }
+        }
     }
 
-    public void depleteHealth(int damage) {
+    public void moveCharacter(int id, float newX, float newY) {
         /*
-         * Decreases player health by damage
-         * */
-        this.character.health -= damage;
+        * Updates the position of the character
+        * */
+        if (verifyId(id)) {
+            this.characterEntities.get(id).position[0] = newX;
+            this.characterEntities.get(id).position[1] = newY;
+        }
     }
 
-    public void increaseLevel() {
+    public void depleteHealth(int id, int damage) {
         /*
-         * Increases the level of enemies by 1 following each wave of enemies.
-         * Need to check if character.team == "Enemy"
+         * Decreases character health by damage
          * */
-        this.character.level += 1;
+        if (verifyId(id)) {
+            this.characterEntities.get(id).health -= damage;
+        }
     }
 
-    public boolean useItem() {
+    public void increaseLevel(int id) {
         /*
-         * Checks if the item is in the characters inventory and then allows them to use it.
-         * This will only be available to the player class, will check insanceof Player.
-         * Returns true iff the player uses the item, false otherwise.
+         * Increases the level of a character following the completion of a wave
          * */
+        if (verifyId(id)) {
+            this.characterEntities.get(id).level += 1;
+        }
+    }
 
-        return true;
+    public boolean canUseItem(int id, String item) {
```

Comment:
Should the type of the second parameter be Item instead of String? @all the other inventory methods